forte/data/data_store.py

```diff
@@ -174,72 +179,109 @@ def add_annotation_raw(
+        r"""This function locates the entry data with `tid` and finds `attr_id`
+        of its attribute `attr_name`. `Tid`, `attr_id`, and `attr_value` are
+        passed to `set_attr()`.
+
+        Args:
+            tid (int): Unique Id of the entry.
+            attr_name (str): Name of the attribute.
+            attr_value (any): Value of the attribute.
+        """
+        if tid not in self.entry_dict:
+            raise KeyError(f"Entry with tid {tid} not found.")
+        entry_type = self.entry_dict[tid][self.entry_type_idx]
+        if attr_name not in self._type_attributes[entry_type]:
+            raise ValueError(f"{entry_type} has no {attr_name} attribute.")
+        attr_id = self._type_attributes[entry_type][attr_name]
+        self.set_attr(tid, attr_id, attr_value)
+
+    def get_attribute(self, tid: int, attr_name: str):
+        r"""This function locates the entry data with `tid` and finds `attr_id`
+        of its attribute `attr_name`. `Tid` and `attr_id` are passed
+        to `get_attr()`.
+
+        Args:
+            tid (int): Unique id of the entry.
+            attr_name (str): Name of the attribute.
+
+        Returns:
+            The value of `attr_name` for the entry with `tid`.
+        """
+        if tid not in self.entry_dict:
+            raise KeyError(f"Entry with tid {tid} not found.")
+        entry_type = self.entry_dict[tid][self.entry_type_idx]
+        if attr_name not in self._type_attributes[entry_type]:
+            raise ValueError(f"{entry_type} has no {attr_name} attribute.")
+        attr_id = self._type_attributes[entry_type][attr_name]
+        return self.get_attr(tid, attr_id)
 
     def delete_entry(self, tid: int):
         r"""This function locates the entry data with `tid` and removes it
         from the data store. This function first removes it from `entry_dict`.
 
         Args:
             tid (int): Unique id of the entry.
-
         """
-        # We retrieve the entry data from entry_dict using tid. We get its
-        # entry type, `type_id`, begin and end indexes. Remove the entry from
-        # `entry_dict` when its information is retrieved. We find the
-        # `entry_type` sortedlist using `type_id`. We bisect the target_list to
+        # We retrieve the entry data from `entry_dict` using tid. We get its
+        # `entry_type`, `type_id`, `begin` and `end` indices. we remove the
+        # entry from `entry_dict` when its information is retrieved. We find the
+        # `entry_type` sortedlist using `type_id`. We bisect the sortedlist to
         # find the index of entry data. Then, `type_id` and index are passed to
         # _delete_entry_by_loc.
         raise NotImplementedError
 
     def _delete_entry_by_loc(self, type_id: int, index_id: int):
-        r"""It removes the index_id'th entry data from the sortedlist of its type.
+        r"""It removes the `index_id`'th entry data from the sortedlist of its type.
```

Comment:
similar, functionality first, then the details. Here the functionality is simply removing an entry of index_id, by taking both the type_id and index_id